routes/volunteering-events.js

```diff
@@ -21,12 +21,29 @@ function getRequestOrOffer(fieldName) {
   };
 }
 
+function getEventsHistory() {
+  return async (ctx) => {
+    try {
+      const queryOptions = DBMethods.processPaginationParameters(ctx);
+      const queryFilter = DBMethods.processDatesRangeParameters(ctx);
+      const eventsHistory = await VolunteeringEvent
+        .find({ ...queryFilter, createdByUserId: ctx.params.id },
+          null,
+          { ...queryOptions, sort: { date: -1 } });
+      DBMethods.handleDocResponse(ctx, eventsHistory);
+    } catch (error) {
+      DBMethods.handleErrors(ctx, error);
+    }
+  };
+}
+
 router.get('/', DBMethods.list(VolunteeringEvent))
   .get('/:id', DBMethods.getById(VolunteeringEvent))
   .get('/:id/request', getRequestOrOffer('request'))
   .get('/:id/offer', getRequestOrOffer('offer'))
-
   .use(auth.authenticate)
+  .use(auth.modificationIsAllowed)
+  .get('/users/:id', getEventsHistory())
```

Comment:
Getting all events for a specific user should be like `/events?userId=id`, if you put it in `/events`.
Useless note: I'd put inside `/users`, but we don't have such routing, so it's irrelevant.

https://github.com/paypal/api-standards/blob/master/api-style-guide.md#filter-a-resource-collection